components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -295,8 +295,10 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
 
                         app.setScopeValidators(filterScopeValidators(application));
 
-                        validateAudiences(application);
-                        app.setAudiences(application.getAudiences());
+                        validateIdTokenAudiences(application);
+                        validateAccessTokenAudiences(application);
+                        app.setIdTokenAudiences(application.getIdTokenAudiences());
+                        app.setAccessTokenAudiences(application.getAccessTokenAudiences());
```

Comment:
we need to make sure to preserve the backword compatibility via a config. based on the boolean flag of the config we need to check which way to go

Reply:
resolved with 08954d00bccda5645fa04905eca48d4cc7ae5f01